File: frontend/src/types/employment.ts
export interface Employment {
  student_no: string
  open_time?: string
  offer_time?: string
  company?: string
  salary?: number
  status?: string
  isdeleted?: number
}

export interface EmploymentCreate {
  student_no: string
  open_time: string
  offer_time: string
  company: string
  salary: number
}

export interface EmploymentUpdate {
  open_time?: string
  offer_time?: string
  company?: string
  salary?: number
  status?: string
}

export interface EmploymentSearch {
  student_no?: string
  company?: string
  min_salary?: number
  max_salary?: number
}

export interface EmploymentSearchResult {
  student_no: string
  student_name: string
  class_no: string
  company: string
  salary: number
  open_time?: string
  offer_time?: string
}

File: sub_module/haohong2/pdc_model/employment.py
from pydantic import BaseModel
from datetime import date
from typing import Optional

from pydantic import BaseModel
from datetime import date
from typing import Optional

# 创建用
class EmploymentCreate(BaseModel):
    student_id: int
    open_time: date
    offer_time: date
    company: str
    salary: int

# 修改用
class EmploymentUpdate(BaseModel):
    open_time: Optional[date] = None
    offer_time: Optional[date] = None
    company: Optional[str] = None
    salary: Optional[int] = None
    status: Optional[str] = None

# 响应体
class EmploymentOut(BaseModel):
    id: int
    student_id: int
    open_time: date
    offer_time: Optional[date]
    company: str
    salary: int

class EmploymentQuery(BaseModel):
    student_id:Optional[int]=None
    company:Optional[str]=None
    min_salary:Optional[int] = None
    max_salary:Optional[int]=None


class EmploymentSearchResponse(BaseModel):
    student_id: int
    student_name: str
    class_id: int
    company: str
    salary: float
    open_time: Optional[date] = None
    offer_time: Optional[date] = None


